axonserver/src/main/java/io/axoniq/axonserver/logging/AuditLog.java

```diff
@@ -0,0 +1,61 @@
+package io.axoniq.axonserver.logging;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationListener;
+import org.springframework.security.authentication.event.AbstractAuthenticationEvent;
+import org.springframework.security.authentication.event.AbstractAuthenticationFailureEvent;
+import org.springframework.security.authentication.event.AuthenticationSuccessEvent;
+import org.springframework.security.authentication.event.InteractiveAuthenticationSuccessEvent;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+
+import java.security.Principal;
+
+@Component
+public class AuditLog implements ApplicationListener<AbstractAuthenticationEvent> {
+
+    public static <T> Logger getLogger() {
+        Class<?> clazz[] = CallingClass.INSTANCE.getCallingClasses();
+
+        return LoggerFactory.getLogger("AUDIT." + clazz[2].getName());
+    }
+
+    public static String enablement(final boolean onOff) {
+        return onOff ? "ENABLED" : "DISABLED";
+    }
+
+    public static String username(Principal principal) {
+        return (principal == null) ? "<anonymous>" : principal.getName();
+    }
+
+    private static final Logger auditLog = getLogger();
+
+    @Override
+    public void onApplicationEvent(AbstractAuthenticationEvent evt) {
```

Comment:
Why is this class in charge of creating/being the audit log as well as logging on an `AbstractAuthenticationEvent`? On face value I'd expect this handler to reside somewhere else.

Reply:
So you'd get a single-method listener class and a statics-only utility class. Fair point.